services/EventTokenService.js

```diff
@@ -0,0 +1,68 @@
+const jwt = require("jsonwebtoken");
+const crypto = require("crypto");
+
+// A service class for Token generation and management
+
+class EventTokenService {
+  static #app_access_key = process.env.LIVE_APP_ACCESS_KEY || config.get("Event100ms.APP_ACCESS_KEY");
+  static #app_secret = process.env.LIVE_APP_SECRET || config.get("Event100ms.APP_SECRET");
+  #managementToken;
+  constructor() {
+    this.#managementToken = this.getManagementToken(true);
+  }
+
+  // A private method that uses JWT to sign the payload with APP_SECRET
+  #signPayloadToToken(payload) {
+    const token = jwt.sign(payload, EventTokenService.#app_secret, {
+      algorithm: "HS256",
+      expiresIn: "24h",
+      jwtid: crypto.randomUUID({ disableEntropyCache: true }),
+    });
+    return token;
+  }
+
+  // A private method to check if a JWT token has expired or going to expire soon
+  #isTokenExpired(token) {
+    try {
+      const { exp } = jwt.decode(token);
+      const buffer = 30; // generate new if it's going to expire soon
+      const currTimeSeconds = Math.floor(Date.now() / 1000);
+      return !exp || exp + buffer < currTimeSeconds;
+    } catch (err) {
+      logger.info("error in decoding token", err);
+      return true;
+    }
+  }
+
+  // Generate new Management token, if expired or forced
+  getManagementToken(forceNew) {
+    if (forceNew || this.#isTokenExpired(this.#managementToken)) {
+      const payload = {
+        access_key: EventTokenService.#app_access_key,
+        type: "management",
+        version: 2,
+        iat: Math.floor(Date.now() / 1000),
+        nbf: Math.floor(Date.now() / 1000),
+      };
+      this.#managementToken = this.#signPayloadToToken(payload);
+    }
+    return this.#managementToken;
+  }
+
+  // Generate new Auth token for a peer
+  getAuthToken({ roomId, userId, role }) {
```

Comment:
Hey @iamitprakash,
As the token is been consumed at the same time, keeping an nbf in itself won't be fruitful because the token will be consumed by that time. As soon as the connection to 100ms is sent, we also add the management token. So, the token is authenticated as soon as it is generated. We can keep the time in nanoseconds but we are not sure about when the token will reach the server. 
If we keep the nbf 2-3 seconds later, users will face the latency which we don't want to happen or joining in the session after 2-3 seconds
If we keep the nbf the 100ms server won't be able to authenticate in case the request goes in & everytime the user has to create a new token. Everytime if the nbf is 2-3 seconds later then the user won't be able to authenticate to 100ms. (Basically, it would delay the token's validity and create unnecessary authentication delays for users.)

So, the usecase in which we keep the nbf to few seconds later won't be a valid case here.